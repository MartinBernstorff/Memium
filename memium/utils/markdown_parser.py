from typing import Protocol

from markdown import Markdown


class MarkdownParser(Protocol):
    def __call__(self, markdown: str) -> str: ...


def to_html(markdown: str) -> str:
    return Markdown(
        output_format="html", extensions=["legacy_em", "fenced_code", "tables", "nl2br"]
    ).convert(markdown)
