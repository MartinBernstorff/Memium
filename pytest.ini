[pytest]
python_files = *.py
python_functions = test_* should_*