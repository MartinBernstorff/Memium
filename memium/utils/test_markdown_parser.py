import pathlib
from dataclasses import dataclass

import pytest

from .markdown_parser import to_html


@dataclass(frozen=True)
class Ex:
    given: str
    # Required setup

    then: str
    # What the expected result is


@pytest.mark.parametrize(
    ("example"),
    [
        Ex("_Method_s", "<em>Method</em>s")  # Example one
    ],
)
def test_markdown_parser(example: Ex):
    assert to_html(example.given) == f"<p>{example.then}</p>"


def test_code_block_parsing(snapshot: str, tmp_path: pathlib.Path):
    contents = pathlib.Path(__file__).parent / "code_block.md"
    parsed = to_html(contents.read_text())
    path = tmp_path / "test.htm"
    path.write_text(parsed)
    assert parsed == snapshot
