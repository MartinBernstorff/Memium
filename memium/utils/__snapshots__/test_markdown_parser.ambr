# serializer version: 1
# name: test_code_block_parsing
  '''
  <pre><code class="language-python">def myfunc():
      return res
  </code></pre>
  '''
# ---
